timesketch/lib/tasks.py

```diff
@@ -71,97 +76,234 @@ def _set_timeline_status(index_name, status, error_msg=None):
+    if not current_app.config.get(u'ENABLE_INDEX_ANALYZERS', False):
+        return None
+
+    for analyzer_name, analyzer_cls in manager.AnalysisManager.get_analyzers():
+        kwarg_list = analyzer_cls.get_kwargs()
+
+        if not analyzer_cls.IS_SKETCH_ANALYZER:
+            if kwarg_list:
+                for kwargs in kwarg_list:
+                    tasks.append(
+                        run_index_analyzer.s(analyzer_name, **kwargs))
+            else:
+                tasks.append(run_index_analyzer.s(analyzer_name))
+
+    return group(tasks)
+
+
+def build_index_pipeline(file_path, timeline_name, index_name, file_extension):
+    """Build a pipeline for index and analysis.
+
+    Args:
+        file_path: Path to the file to index.
+        timeline_name: Name of the timeline to create.
+        index_name: Name of the index to index to.
+        file_extension: The file extension of the file.
+
+    Returns:
+        Celery chain with indexing task and analyzer task group.
+    """
+    index_task_class = _get_index_task_class(file_extension)
+    analyzer_task_group = _get_index_analyzers()
+
+    index_task = index_task_class.s(
+        file_path, timeline_name, index_name, file_extension)
+
+    # If there are no analyzers just run the indexer.
+    if not analyzer_task_group:
+        return index_task
+
+    return chain(index_task, analyzer_task_group)
+
+
+def build_sketch_analysis_pipeline(sketch_id, index_name):
+    """Build a pipeline for sketch analysis.
+
+    Args:
+        sketch_id: The ID of the sketch to analyze.
+        index_name: Elasticsearch index name.
+
+    Returns:
+        Celery group with analysis tasks or None if not analyzers are enabled.
+    """
+    tasks = []
+
+    # Exit early if sketch analyzers are disabled.
+    if not current_app.config.get(u'ENABLE_SKETCH_ANALYZERS', False):
+        return None
+
+    for analyzer_name, analyzer_cls in manager.AnalysisManager.get_analyzers():
+        kwarg_list = analyzer_cls.get_kwargs()
+
+        if not analyzer_cls.IS_SKETCH_ANALYZER:
+            continue
+
+        if kwarg_list:
+            for kwargs in kwarg_list:
+                tasks.append(run_sketch_analyzer.s(
+                    sketch_id, index_name, analyzer_name, **kwargs))
+        else:
+            tasks.append(run_sketch_analyzer.s(
+                sketch_id, index_name, analyzer_name))
+
+    return group(tasks)
+
+
+@celery.task(track_started=True)
+def run_index_analyzer(index_name, analyzer_name, **kwargs):
+    """Create a Celery task for an index analyzer.
+
+    Args:
+      index_name: Name of the datastore index.
+      analyzer_name: Name of the analyzer.
+
+    Returns:
+      Name (str) of the index.
+    """
+    analyzer_class = manager.AnalysisManager.get_analyzer(analyzer_name)
+    analyzer = analyzer_class(index_name=index_name, **kwargs)
+    result = analyzer.run_wrapper()
+    logging.info('[%s] result: %s' % (analyzer_name, result))
```

Comment:
Any reason you're not using `{0:s}` and `.format()` here?  I thought it might be because it's a logging line, but there are instances of using `format()` in logging lines in this PR.